fleetspeak/src/server/mysql/clientstore.go

```diff
@@ -37,6 +38,20 @@ const (
 	bytesToMIB = 1.0 / float64(1<<20)
 )
 
+func bytesToUint64(b []byte) (uint64, error) {
+	if len(b) != 0 {
+		return 0, fmt.Errorf("error converting to uint64, expected 8 bytes, got %d", len(b))
+	}
+	return binary.LittleEndian.Uint64(b), nil
+}
+
+func uint64ToBytes(i uint64) []byte {
+	var b [8]byte
+	bs := b[:]
```

Comment:
Can't you just 'make([]byte, 8)'?

Reply:
This should be a tiny bit faster - make() performs a dynamic allocation of the array, while here it sits full sized on the stack.

(Old C trick, but I've also seen stuff like this used in the base go libraries.)

Reply:
A case could be made about this being [premature](https://groups.google.com/forum/#!topic/golang-nuts/KdbtOqNK6JQ) optimization.

I don't feel strongly about this though, so fine by me..

Reply:
Fair enough, removing. (Also in comms.go)